File: app/src/main/java/com/tokeninc/sardis/application_template/helpers/PrintHelpers/SalePrintHelper.java
package com.tokeninc.sardis.application_template.helpers.PrintHelpers;
/*
import android.content.Context;

import com.token.printerlib.PrinterDefinitions;
import com.token.printerlib.PrinterDefinitions.Alignment;
import com.token.printerlib.StyledString;
import com.tokeninc.deviceinfo.DeviceInfo;
import com.tokeninc.sardis.application_template.AppTemp;
import com.tokeninc.sardis.application_template.Entity.SampleReceipt;
import com.tokeninc.sardis.application_template.Entity.SlipType;
import com.tokeninc.sardis.application_template.R;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;

public class SalePrintHelper extends BasePrintHelper{

    public static String getFormattedText(SampleReceipt receipt, SlipType slipType, Context context, Integer ZNO, Integer ReceiptNo)
    {
        StyledString styledText = new StyledString();


        if(slipType == SlipType.CARDHOLDER_SLIP){
            if(!((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name())  && !((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.VUK507.name())){
                printSlipHeader(styledText, receipt);
            }
        }
        else {
            printSlipHeader(styledText, receipt);
        }

        styledText.newLine();
        if (slipType == SlipType.CARDHOLDER_SLIP) {
            styledText.addTextToLine("MÜŞTERİ NÜSHASI", Alignment.Center);
            styledText.newLine();
        }
        else if (slipType == SlipType.MERCHANT_SLIP) {
            styledText.addTextToLine("İŞYERİ NÜSHASI", Alignment.Center);
            styledText.newLine();
        }
        styledText.addTextToLine("SATIŞ", Alignment.Center);

        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yy HH:mm:ss", Locale.getDefault());
        String time = sdf.format(Calendar.getInstance().getTime());

        styledText.newLine();

        if(slipType == SlipType.CARDHOLDER_SLIP){
            if(((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name())|| ((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.VUK507.name())){
                styledText.addTextToLine("C ONLINE", Alignment.Center);
            }
            else{
                styledText.addTextToLine(time + " " + "C ONLINE", Alignment.Center);
            }
        }
        else if(slipType == SlipType.MERCHANT_SLIP){
            styledText.addTextToLine(time + " " + "C ONLINE", Alignment.Center);
        }

        styledText.newLine();
        styledText.addTextToLine(receipt.getCardNo(), Alignment.Center);

        styledText.newLine();
        styledText.addTextToLine(receipt.getFullName(), Alignment.Center);

        styledText.setLineSpacing(1f);
        styledText.setFontSize(14);
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold);
        styledText.newLine();
        styledText.addTextToLine("TUTAR:");
        styledText.addTextToLine(receipt.getAmount(), Alignment.Right);

        styledText.setLineSpacing(0.5f);
        styledText.setFontSize(10);
        styledText.newLine();
        if (slipType == SlipType.CARDHOLDER_SLIP) {
            styledText.addTextToLine("KARŞILIĞI MAL/HİZM ALDIM", Alignment.Center);
        }
        else {
            styledText.addTextToLine("İşlem Şifre Girilerek Yapılmıştır", Alignment.Center);
            styledText.newLine();
            styledText.addTextToLine("İMZAYA GEREK YOKTUR", Alignment.Center);
        }

        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Bold);
        styledText.setFontSize(12);
        styledText.newLine();
        styledText.addTextToLine("SN: " + receipt.getSerialNo());
        styledText.addTextToLine("ONAY KODU: " + receipt.getApprovalCode(), Alignment.Right);

        styledText.setFontSize(8);
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold);
        styledText.newLine();
        styledText.addTextToLine("GRUP NO:" + receipt.getGroupNo());

        styledText.newLine();
        styledText.addTextToLine("AID: " + receipt.getAid());

        if (slipType == SlipType.MERCHANT_SLIP) {
            addTextToNewLine(styledText, "*MALİ DEĞERİ YOKTUR*", Alignment.Center, 8);
        }

        if (slipType == SlipType.MERCHANT_SLIP) {
            if(((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name())) {
                styledText.newLine();
                styledText.addTextToLine("Z NO: " +ZNO, Alignment.Right);
                styledText.addTextToLine("FİŞ NO: " +ReceiptNo, Alignment.Left);
            }
        }

        if (slipType == SlipType.MERCHANT_SLIP) {
            if(((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name())
                    || ((AppTemp) context.getApplicationContext()).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.VUK507.name())) {

            addTextToNewLine(styledText, ((AppTemp) context.getApplicationContext()).getCurrentFiscalID(), Alignment.Center, 8);

            }
        }

        styledText.newLine();
        styledText.addTextToLine("BU İŞLEM YURT İÇİ KARTLA YAPILMIŞTIR", Alignment.Center);
        styledText.newLine();
        styledText.addTextToLine("BU BELGEYİ SAKLAYINIZ", Alignment.Center);
        styledText.newLine();

        styledText.printBitmap("ykb", 20);
        styledText.addSpace(100);
        return styledText.toString();
    }

    public static byte[] getBitmap(Context context) {
        return getBitmapReceiptArray(context, R.raw.ziraat_fis);
    }

    public static byte[] getBitmapReceiptArray(Context context, int resourceId) {

        byte[] bitmap;
        try
        {
            InputStream inStream = context.getResources().openRawResource(resourceId);
            bitmap = new byte[inStream.available()];

            ByteArrayOutputStream baos = new ByteArrayOutputStream();
            byte[] buff = new byte[10240];
            int i = Integer.MAX_VALUE;

            while ((i = inStream.read(buff, 0, buff.length)) > 0) {
                baos.write(buff, 0, i);
            }

            bitmap = baos.toByteArray(); // be sure to close InputStream in calling function

            inStream.close();
        }
        catch (IOException e)
        {
            bitmap = null;
            e.printStackTrace();
        }

        return bitmap;
    }

}

 */

File: app/src/main/java/com/tokeninc/sardis/application_template/helpers/printHelpers/SalePrintHelper.kt
package com.tokeninc.sardis.application_template.helpers.printHelpers

import android.content.ContentValues
import android.content.Context
import com.token.printerlib.PrinterDefinitions
import com.token.printerlib.StyledString
import com.tokeninc.deviceinfo.DeviceInfo
import com.tokeninc.sardis.application_template.AppTemp
import com.tokeninc.sardis.application_template.OnlineTransactionResponse
import com.tokeninc.sardis.application_template.SampleReceipt
import com.tokeninc.sardis.application_template.database.transaction.TransactionCol
import com.tokeninc.sardis.application_template.enums.SlipType
import com.tokeninc.sardis.application_template.helpers.StringHelper
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.text.SimpleDateFormat
import java.util.*

class SalePrintHelper: BasePrintHelper() {
    //TODO  MASKCARDNUMBER yap, YKB slibi aynısı, application conxtext currentDevice'a göre slip


    fun getFormattedText(slipType: SlipType, contentValues: ContentValues, onlineTransactionResponse: OnlineTransactionResponse,
                         context: Context, ZNO: Int, ReceiptNo: Int, isCopy: Boolean): String {
        val styledText = StyledString()
        val stringHelper = StringHelper()
        if (slipType === SlipType.CARDHOLDER_SLIP) {
        if (!(context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name) &&
            !(context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.VUK507.name))  {
                //printSlipHeader(styledText, contentValues) //TODO Burada merchantID, TerminalID, PosID parametre olarak gitmeli
            }
        } else {
            //printSlipHeader(styledText, receipt)
        }
        styledText.newLine()
        if (slipType === SlipType.CARDHOLDER_SLIP) {
            styledText.addTextToLine("MÜŞTERİ NÜSHASI", PrinterDefinitions.Alignment.Center)
            styledText.newLine()
        } else if (slipType === SlipType.MERCHANT_SLIP) {
            styledText.addTextToLine("İŞYERİ NÜSHASI", PrinterDefinitions.Alignment.Center)
            styledText.newLine()
        }
        styledText.addTextToLine("SATIŞ", PrinterDefinitions.Alignment.Center)
        val sdf = SimpleDateFormat("dd-MM-yy HH:mm:ss", Locale.getDefault())
        val time = sdf.format(Calendar.getInstance().time)
        styledText.newLine()
        if (slipType === SlipType.CARDHOLDER_SLIP) {
         if ((context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name) ||
             (context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.VUK507.name))
         {
                styledText.addTextToLine("C ONLINE", PrinterDefinitions.Alignment.Center)
            } else {
                styledText.addTextToLine("$time C ONLINE", PrinterDefinitions.Alignment.Center)
            }
        } else if (slipType === SlipType.MERCHANT_SLIP) {
            styledText.addTextToLine("$time C ONLINE", PrinterDefinitions.Alignment.Center)
        }

        styledText.newLine()
        styledText.addTextToLine(stringHelper.MaskTheCardNo(contentValues.getAsString(TransactionCol.Col_PAN.name)), PrinterDefinitions.Alignment.Center)
        styledText.newLine()
        if (contentValues.get(TransactionCol.Col_CustName.name) != null){
            styledText.addTextToLine(contentValues.getAsString(TransactionCol.Col_CustName.name), PrinterDefinitions.Alignment.Center)
        }
        styledText.setLineSpacing(1f)
        styledText.setFontSize(14)
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold)
        styledText.newLine()
        styledText.addTextToLine("TUTAR:")
        styledText.addTextToLine(stringHelper.getAmount(contentValues.getAsString(TransactionCol.Col_Amount.name).toInt()), PrinterDefinitions.Alignment.Right)
        styledText.setLineSpacing(0.5f)
        styledText.setFontSize(10)
        styledText.newLine()
        if (slipType === SlipType.CARDHOLDER_SLIP) {
            styledText.addTextToLine(
                "KARŞILIĞI MAL/HİZM ALDIM",
                PrinterDefinitions.Alignment.Center
            )
        } else {
            styledText.addTextToLine(
                "İşlem Şifre Girilerek Yapılmıştır",
                PrinterDefinitions.Alignment.Center
            )
            styledText.newLine()
            styledText.addTextToLine("İMZAYA GEREK YOKTUR", PrinterDefinitions.Alignment.Center)
        }
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Bold)
        styledText.setFontSize(12)
        styledText.newLine()
        styledText.addTextToLine("SN: " + contentValues.getAsString(TransactionCol.Col_STN.name))
        styledText.addTextToLine(
            "ONAY KODU: " + contentValues.getAsString(TransactionCol.Col_AuthCode.name),
            PrinterDefinitions.Alignment.Right
        )
        styledText.setFontSize(8)
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold)
        styledText.newLine()
        styledText.addTextToLine("GRUP NO:" + contentValues.getAsString(TransactionCol.Col_GUP_SN.name))
        styledText.newLine()
        styledText.addTextToLine("AID: " + contentValues.getAsString(TransactionCol.Col_Aid.name))
        if (slipType === SlipType.MERCHANT_SLIP) {
            addTextToNewLine(
                styledText,
                "*MALİ DEĞERİ YOKTUR*",
                PrinterDefinitions.Alignment.Center,
                8
            )
        }
        if (slipType === SlipType.MERCHANT_SLIP) {
           if ((context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name)
            ) {
                styledText.newLine()
                styledText.addTextToLine("Z NO: $ZNO", PrinterDefinitions.Alignment.Right)
                styledText.addTextToLine("FİŞ NO: $ReceiptNo", PrinterDefinitions.Alignment.Left)
            }
        }
        if (slipType === SlipType.MERCHANT_SLIP) {
            if ((context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.ECR.name)
                || (context.applicationContext as AppTemp).getCurrentDeviceMode().equals(DeviceInfo.PosModeEnum.VUK507.name)
            ) {
                addTextToNewLine(styledText, (context.applicationContext as AppTemp).getCurrentFiscalID(),
                    PrinterDefinitions.Alignment.Center, 8)
            }
        }
        styledText.newLine()
        styledText.addTextToLine(
            "BU İŞLEM YURT İÇİ KARTLA YAPILMIŞTIR",
            PrinterDefinitions.Alignment.Center
        )
        styledText.newLine()
        styledText.addTextToLine("BU BELGEYİ SAKLAYINIZ", PrinterDefinitions.Alignment.Center)
        styledText.newLine()
        styledText.printBitmap("ykb", 20)
        styledText.addSpace(100)
        return styledText.toString()
    }
}